atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepableTimestamps.java

```diff
@@ -0,0 +1,137 @@
+
+    public SweepableTimestamps(KeyValueService kvs, WriteInfoPartitioner partitioner) {
+        super(kvs, TargetedSweepTableFactory.of().getSweepableTimestampsTable(null).getTableRef(), partitioner);
+    }
+
+    @Override
+    void populateCells(PartitionInfo info, List<WriteInfo> writes, Map<Cell, byte[]> cellsToWrite) {
+        SweepableTimestampsTable.SweepableTimestampsRow row = computeRow(info);
+        SweepableTimestampsTable.SweepableTimestampsColumn col = computeColumn(info);
+
+        SweepableTimestampsTable.SweepableTimestampsColumnValue colVal =
+                SweepableTimestampsTable.SweepableTimestampsColumnValue.of(col, DUMMY);
+
+        cellsToWrite.put(SweepQueueUtils.toCell(row, colVal), colVal.persistValue());
+    }
+
+    Optional<Long> nextSweepableTimestampPartition(ShardAndStrategy shardStrategy, long lastSweptFine, long sweepTs) {
+        long minFineExclusive = lastSweptFine;
+        long maxFineExclusive = SweepQueueUtils.tsPartitionFine(sweepTs);
+        return nextSweepablePartition(shardStrategy, minFineExclusive + 1, maxFineExclusive);
+    }
+
+    void deleteRow(ShardAndStrategy shardStrategy, long partitionFine) {
+        byte[] row = computeRow(shardStrategy, SweepQueueUtils.partitionFineToCoarse(partitionFine)).persistToBytes();
+
+        RangeRequest request = RangeRequest.builder()
+                .startRowInclusive(row)
+                .endRowExclusive(RangeRequests.nextLexicographicName(row))
+                .retainColumns(ColumnSelection.all())
+                .build();
+
+        deleteRange(request);
+    }
+
+    private Optional<Long> nextSweepablePartition(ShardAndStrategy shardAndStrategy, long minFine, long maxFine) {
+        Optional<ColumnRangeSelection> range = getColRangeSelection(minFine, maxFine);
+
+        if (!range.isPresent()) {
+            return Optional.empty();
+        }
+
+        long current = SweepQueueUtils.partitionFineToCoarse(minFine);
+        long maxCoarse = SweepQueueUtils.partitionFineToCoarse(maxFine);
+
+        while (current <= maxCoarse) {
+            Optional<Long> candidateFine = getCandidatesInCoarsePartition(shardAndStrategy, current, range.get());
+            if (candidateFine.isPresent()) {
+                return candidateFine;
+            }
+            current++;
+        }
+        return Optional.empty();
+    }
+
+    private Optional<ColumnRangeSelection> getColRangeSelection(long minFine, long maxFine) {
+        if (minFine >= maxFine) {
+            return Optional.empty();
+        }
+        byte[] start = SweepableTimestampsTable.SweepableTimestampsColumn.of(minFine).persistToBytes();
+        byte[] end = SweepableTimestampsTable.SweepableTimestampsColumn.of(maxFine).persistToBytes();
+        return Optional.of(new ColumnRangeSelection(start, end));
+    }
+
+    private Optional<Long> getCandidatesInCoarsePartition(ShardAndStrategy shardStrategy, long partitionCoarse,
```

Comment:
would this be more accurately called `getNextCandidateInCoarsePartition`?

Reply:
Renamed in next PR

Reply:
Noting that this isn't in part 9. I don't care that much, though.